### ISSUE TITLE
Non-Steam GOG game "Blood Omen: Legacy of Kain" not working

### ISSUE THREAD

**Issue by folavo** [sonic2kk/steamtinkerlaunch#628]
First of all, thanks a lot all your work.

## System Information
- SteamTinkerLaunch version: v11.11
- Distribution: SteamOS 3.3.2
- Installation Method: ProtonUp-Qt

## Issue Description
Trying to start "Non-Steam" games will result in a hanging steam icon (nothing happens). I eventually abort the game to get back to the Menu.
I have tried adding the game from inside Desktop Steam and through "steamtinkerlaunch ansg", but result is the same.
Have also tried changing between different Proton versions and result is the same.

Let me know how else I can help.

Logs attached.

## Logs
[gamelaunch_id_3920664057.log](https://github.com/frostworx/steamtinkerlaunch/files/9758164/gamelaunch_id_3920664057.log)
[steamtinkerlaunch.log](https://github.com/frostworx/steamtinkerlaunch/files/9758166/steamtinkerlaunch.log)
[steamtinkerlaunch_id_3920664057.log](https://github.com/frostworx/steamtinkerlaunch/files/9758218/steamtinkerlaunch_id_3920664057.log)

**Comment by sonic2kk** [comment]
Just a note for clarity (mainly for others looking at this issue), STL will automatically update to the latest git release of STL. So this issue should currently be present on master.

@folavo Haven't dug into the logs yet but have you verified that this application definitely works on Steam Deck/Linux, independent of STL?

Also, just to confirm, you're setting STL as a compatibility tool for a Non-Steam Game? This is not related to launching a regular game with STL and setting a custom command with STL?

**Comment by sonic2kk** [comment]
Line of interest from `gamelaunch` log:

```
[S_API FAIL] SteamAPI_Init() failed; no appID found.
Either launch the game from Steam, or put the file steam_appid.txt containing the correct appID in your game folder
```

The steamtinkerlaunch log is very strange, particular lines of note:

```
Tue Oct 11 06:40:31 PM CEST 2022 INFO - getCurrentCommandline - No Proton in command line arguments '-version'
...
Tue Oct 11 06:40:32 PM CEST 2022 INFO - main - Initializing first Proton:
Tue Oct 11 06:40:32 PM CEST 2022 INFO - initOldProtonArr - Creating an initial array with available Proton versions using the file '/dev/shm/steamtinkerlaunch/ProtonCSV.txt' which was created during a previous run
Tue Oct 11 06:40:32 PM CEST 2022 INFO - initFirstProton - Initializing Proton
Tue Oct 11 06:40:32 PM CEST 2022 INFO - initFirstProton - Initial Proton version 'FIRSTUSEPROTON' from '/home/deck/.config/steamtinkerlaunch/default_template.conf' is 'proton-7.0-4'
...
Tue Oct 11 06:40:32 PM CEST 2022 INFO - main - Checking command line: incoming arguments '-version'
Tue Oct 11 06:40:32 PM CEST 2022 INFO - commandline ------------------------
Tue Oct 11 06:40:33 PM CEST 2022 INFO - commandline - arg1 '-version' is no valid command
```

Not sure where the heck the `-version` line is coming from at the end of the steamtinkerlaunch log,but I suspect this is the issue.

Is there by any chance a `-version` in the "launch options" section of the application? Or do you know of somewhere else that this `-version` flag may be coming from?

**Comment by frostworx** [comment]
Thanks for a proper and good bug report, @folavo
I guess the `-version` log can be ignored and is not related to the reported issue. probably just generated when trying to get the current steamtinkerlaunch version from command line.
the 3920664057 log is fine though, so no need for an other log.

The actual launch command looks fine so far, the only problem I see now is that steamtinkerlaunch times out with waiting for a game window and therefore exists "gracefully".
Does the Kain.exe work when starting as non-steam game without steamtinkerlaunch. If so, using the same proton version and can you estimate how long it takes until it opens a window?

**Comment by frostworx** [comment]
oh, I haven't seen @sonic2kk's comment -sorry. apparently that is the problem. It is a steam game and wants to be opened as valid steam game from steam directly. Seems like a wontfix and not related to steamtinkerlaunch then

**Comment by sonic2kk** [comment]
> Seems like a wontfix and not related to steamtinkerlaunch then

I guess that might depend on whether or not it works without STL. If it only fails when STL is selected as the compatibility tool there might be something wrong here. Still not sure if it launches at all from Steam without STL.

The line in the Proton launch log I guess it is, does look like an error directly from Steam. But STL's log also looks a bit strange, or more that it looks very empty? Is this expected? Maybe if it's Steam itself that fails to run the game, that's why it's so thin compared to a "normal" log?

Also OP, it would be good to know if this affects more than this specific Non-Steam game.

**Comment by frostworx** [comment]
hm? why "very empty"? not sure if I understand you correctly here. In case you missed it, I think you can ignore the steamtinkerlaunch.log because it is likely invalid/not related. check the steamtinkerlaunch_id_3920664057.log instead.

Generally: it is indeed possible that non-steam windows games have an issue, because I haven't tried this for a very long time, and can't remember any useful report with similar issues.
Here, the direct complain from the game that it wants to be started via steam is pretty obvious though. the Steam DRM just says no for a reason.

**Comment by sonic2kk** [comment]
> In case you missed it, I think you can ignore the steamtinkerlaunch.log because it is likely invalid/not related

Ahh okay, I got you now.

Yes, it does seem like the complaint in the game's log (probably generated by Proton here?) is related to Steam's DRM. There is no game name or any other useful hints anywhere in the log it looks like, so @folavo I think some more information would be useful here:

- What custom game are you trying to launch? Can you give us any background here?
- Does this program work with Wine/Proton in general?
- Does this program work without SteamTinkerLaunch?

> Generally: it is indeed possible that non-steam windows games have an issue

I am only one user and I haven't actually tried on Deck, but on my PC I've tested a few Non-Steam Windows games and they work fine (just small single-exe games from Itch.io, nothing major so haven't tested too extensively for any bigger  games that might require more workarounds that STL does not provide)

<hr>

Interestingly, STL's version is set to `v11.11` in the STL log (I know you said to ignore it but I still had it open and just noticed it). Seems a little strange to me but probably unrelated.

**Comment by frostworx** [comment]
the game seems to be https://store.steampowered.com/app/242960/Blood_Omen_2_Legacy_of_Kain/
likely not owned in the steam library _(wouldn't make sense to start it as non steam game)_, which causes the error.
Maybe the game was started without internet connection and therefore could not update to lastest git master

**Comment by sonic2kk** [comment]
Good find!

A few things to note:
- Lack of internet connection is a good call, and leads into my next point:
- Some games require an Internet connection to run (especially games with DRM and games from Square Enix in general in my experience), a good way to find this out is under the Steam Deck status box, however...
- This game is marked as unsupported by Valve on Steam Deck, but [ProtonDB says otherwise](https://www.protondb.com/app/242960)
- Starting this as a Non-Steam game when it is not owned on Steam could be causing problems. Would be good to know two things:
  - If this game was purchased "legally" and just on another storefront (like on [GOG](https://www.gog.com/en/game/legacy_of_kain_blood_omen_2)) (the title is formatted differently between GOG and Steam, but it appears to be the same game)
  - If this game works without STL

If this is the GOG version of the game, and if it works as a Steam shortcut normally and only causes problems when attempting to use STL, that would be strange. It would also be a bit strange for Steam DRM to flag this up.

It might be possible to create a file called `steam_appid.txt` in the game's exe folder with the value `242960` (the game's AppID). Then try running again and see if that pleases the Steam DRM.

And my request for more context around how OP is trying to run the game still stands

**Comment by frostworx** [comment]
good summary 👍.

I'd like to have more context as well to proceed here.

```
[S_API FAIL] SteamAPI_Init() failed; no appID found.
Either launch the game from Steam, or put the file steam_appid.txt containing the correct appID in your game folder.
```
Is likely coming from the game-bundled steamapi and therefore this is a steam and no gog copy.
I own the game, so it would be pretty easy to find out if it works on the steamdeck.

Good hint regarding `steam_appid.txt`. In fact steamtinkerlaunch has an option to create that file automatically :)

**Comment by frostworx** [comment]
The _(owned!)_ game works ootb with and without `steamtinkerlaunch` on the steam deck

**Comment by frostworx** [comment]
ah, the request came from here btw: https://www.reddit.com/r/SteamTinkerLaunch/comments/y0gwna/cant_run_nonsteam_games/

**Comment by sonic2kk** [comment]
I actually just noticed that I own it too (have 1200+ games, most I don't even know I own...) and was about to test but you beat me to it :laughing: 

Do you know if it works without an Internet connection? Some games have weird DRM that expects an Internet connection but for a game from 2002 and also available on GOG, I'm not sure. Maybe the internet connection has no bearing here for the game at all, I just know some Square Enix games expect it.

EDIT: Very outlandish thought I had: If the game was owned on Steam and launched through STL previously, and then later a Non-Steam version was attempted, could there be any sort of conflict here? E.g., if STL reads some of the exe info the same. I have no idea here, it was just a thought I had *just* in case - Though with my understanding of where and how STL stores config files I think that this is not an issue - and if it was, we would see more things and/or other things going wrong.

**Comment by folavo** [comment]
Hey,

So, the game was bought in GOG - it's definitely not a steam game.
It works perfectly (including videos) with Proton-7.2-GE-2

When launching through Steam directly, it takes around 2-3 seconds.

Please note I don't even find it in Steam. This is the "Legacy of Kain: Blood Omen" game.

**Comment by frostworx** [comment]
Yeah, it starts fine without network as well. I have >6600 games on steam and don't know most of them as well :)
My favorite game was collecting games for a long time, I mostly beat it meanwhile fortunately _(only some achievements to unlock once in a while)_ 😀

**Comment by sonic2kk** [comment]
>  It works perfectly (including videos) with Proton-7.2-GE-2

This is an extremely old version of Proton, have you tried with a newer version? I doubt it but maybe this is causing issues with STL (I know the versions named like GE-Proton have changed the Wine version and such). Though you mentioned changing Proton versions, I guess you tried Valve's Proton 7.0-4?

> Please note I don't even find it in Steam. This is the "Legacy of Kain: Blood Omen" game.

Ah, I see now. It's [this game](https://www.gog.com/en/game/blood_omen_legacy_of_kain) I guess?

So the issue here seems to be launching a game on GOG through STL. Seems very strange, don't think I've ever tried this and since the game works without a network connection and is available from GOG I would say this isn't a DRM issue (maybe Steam DRM, but not any weird custom Square Enix DRM or anything)

Seems like there might be some SteamTinkerLaunch-specific problem here. Note that in the STL Log it seems use the default AppID.

**Comment by folavo** [comment]
Hi guys, I have noticed you have referenced the wrong game in your comments. This is not "Blood Omen 2" ... this is the original "Blood Omen" which does not exist in Steam

**Comment by frostworx** [comment]
ah, part one then _(put it on my gog wishlist, not buying it full price for a quick test 😀)_
can you post a full recurive game dir list? I'm curious where that steam drm comes from.

**Comment by frostworx** [comment]
> Note that in the STL Log it seems use the default AppID.

where? it uses `3920664057`

**Comment by frostworx** [comment]
Ah I just recognized you enabled multiple options. I'd start with disabling them _(starting with FlawlessWidescreen)_
very likely one of the enabled options doesn't work with the game

**Comment by sonic2kk** [comment]
In the `steamtinkerlaunch.log` file it seems like it's using the Placeholder AppID. I can't find `3920664057` anywhere in the `steamtinkerlaunch.log`, but it also doesn't have any game run information.

```
Tue Oct 11 06:40:31 PM CEST 2022 START - ######### Initializing Game Launch 31337 using SteamTinkerLaunch v11.11 #########
```

EDIT: Ohhh I see now, this log file might be from adifferent launch. AH! I understand what you meant earlier, the log starting with steamtinkerlaunch_id_<id>.log` is the actual game run log. I thought this was an alias to the Proton log, my bad.

**Comment by frostworx** [comment]
> In the `steamtinkerlaunch.log` file it seems like it's using the Placeholder AppID. I can't find `3920664057` anywhere in the `steamtinkerlaunch.log`, but it also doesn't have any game run information.
> 
> ```
> Tue Oct 11 06:40:31 PM CEST 2022 START - ######### Initializing Game Launch 31337 using SteamTinkerLaunch v11.11 #########
> ```
> 
> EDIT: Ohhh I see now, this log file might be from adifferent launch

that's what I was trying to say twice above 😀

**Comment by sonic2kk** [comment]
Yeah it definitely looks like a lot of tweaks are made here with STL - Including most notably some winetricks (may be incompatible in various ways, newer Winetricks version `20220411-next` might not be compatible with such an old Proton version, may break the game, etc etc). Though if it works without STL it could be some of the custom settings here that are breaking it, like FlawlessWidescreen (I think the prefix used by STL and by Steam to run the game should be  the same)

Probably worth disabling the custom options and/or starting with a clean wine prefix - You should be able to access it from Desktop Mode by clicking on "Game Files" in the SteamTinkerLaunch Main Menu and selecting `WINEPREFIX`. This should open the folder to the prefix, for you, you'll want to delete the folder `3920664057`. That'll clear out any Winetricks too and start the game as if it was a fresh install

**Comment by folavo** [comment]
Tried cleaning the winprefix. and removing all the selected options - result is the same.

> Ah, I see now. It's [this game](https://www.gog.com/en/game/blood_omen_legacy_of_kain) I guess?
@frostworx I can donate you the game for your support... if you don't have the game, you should anyway! no strings attached
If you are OK, share me an email where I can send it.

**Comment by frostworx** [comment]
@folavo this is very generous from you, thanks a lot for the offer! But it is really not necessary!
If you can and if the listing doesn't contain any personal info (unlikely, but better safe than sorry) it would be nice if you would attach a list of the game dir though.
Please also attach a fresh log from your new attempt, to make sure all optional features were disabled correctly (it could still be possible that you enabled them in your default config, which would mean that they are automatically enabled when you start with a fresh game config)

**Comment by folavo** [comment]
Here are the new logs ... unfortunately I'm a bit tired and did the mistake to overwrite one of the logs since they had the same name.
[3365402217.log](https://github.com/frostworx/steamtinkerlaunch/files/9759268/3365402217.log)
[ls -R.txt](https://github.com/frostworx/steamtinkerlaunch/files/9759269/ls.-R.txt)
[steamtinkerlaunch.log](https://github.com/frostworx/steamtinkerlaunch/files/9759270/steamtinkerlaunch.log)

Hope you can do with these, otherwise I'll have to re-do the test

**Comment by frostworx** [comment]
thanks! unfortunately the file list doesn't has any steam related files.
Please grab another steamtinkerlaunch.log.
the one you attached is one of those which created, when Steam updates its cache in the background _(`run /home/deck/.local/share/Steam/ubuntu12_32/../bin/d3ddriverquery64.exe -d3d12`)_

**Comment by folavo** [comment]
I reinstalled both the game and steamtinker from scratch and ran the game using default options (didn't touch any option). note: the result is the almost the same - but it crashes alone instead of hanging indefinitely. 

Logs attached. I noticed there is a weird 31337.log which ĩ don't know what it is, so I copy it as well.
[gamelaunch_2511443424.log](https://github.com/frostworx/steamtinkerlaunch/files/9762305/gamelaunch_2511443424.log)
[steamtinkerlaunch_31337.log](https://github.com/frostworx/steamtinkerlaunch/files/9762306/steamtinkerlaunch_31337.log)
[steamtinkerlaunch_2511443424.log](https://github.com/frostworx/steamtinkerlaunch/files/9762307/steamtinkerlaunch_2511443424.log)
[steamtinkerlaunch.log](https://github.com/frostworx/steamtinkerlaunch/files/9762308/steamtinkerlaunch.log)

**Comment by frostworx** [comment]
from the [gamelaunch_2511443424.log](https://github.com/frostworx/steamtinkerlaunch/files/9762305/gamelaunch_2511443424.log):

`Proton: Upgrading prefix from 7.2-GE-1 to 7.0-100 (/home/deck/.local/share/Steam/steamapps/compatdata/2511443424/)
Proton: Prefix has an invalid version?! You may want to back up user files and delete this prefix.`

@ "weird 31337.log" : https://github.com/frostworx/steamtinkerlaunch/issues/628#issuecomment-1275595180

@ [steamtinkerlaunch.log](https://github.com/frostworx/steamtinkerlaunch/files/9762308/steamtinkerlaunch.log):
 https://github.com/frostworx/steamtinkerlaunch/issues/628#issuecomment-1275595180

@ steamtinkerlaunch_2511443424.log:
the log looks fine so far. Seems like the game window id was even found and the game crashed afterwards.
it is very unlikely that a wine/game crash is somehow steamtinkerlaunch related (though not 100% impossible).
I'd start with a clean compatdata, maybe it already helps.
for a proper test I also suggest trying the same proton version used to create the new compatdata without steamtinkerlaunch.

**Comment by frostworx** [comment]
https://www.gog.com/forum/blood_omen_legacy_of_kain/linux_users_and_tips
might have several useful hints

**Comment by frostworx** [comment]
just downloaded the iso from archive.org, added the setup.exe as non-steam game, 
started via steamtinkerlaunch (using GE-Proton7-37)
installed it _(had to set z: to the game dir using winecfg and had to rename multiple files to uppercase, also installed all files from the mounted iso into the install dest dir as the installer failed mostly - because of case sensitive code - it is a mess)_.
changed the exe to the installed kain.exe afterwards
ddraw native, buillin override didn't help, the game crashed anyway .
installed the Veroks Patch mentioned above and the game justj works using steamtinkerlaunch.

edit: [here](https://www.gog.com/forum/blood_omen_legacy_of_kain/veroks_gl_wrapper_patch/post32) is an active discussion about the patch (haven't really read it in detail buit might have some interesting information as well)
deleting the game from my system now again

**Comment by folavo** [comment]
Ok. will try with the GLWrapper.
thx

**Comment by folavo** [comment]
@frostworx the patch doesn't work with my GOG files - I tried just copying the files over the installation and the videos became corrupted. 
So, I went and grabbed myself a ISO in the internet (I own the game anyway) and from this I could install the patch (albeit with all the challenges you mentioned). Now the game at least was playable in Windows and I went to copy the contents to Steam Deck: here is where I found a new issue: it asks for the CD-ROM to be inserted :-( oh god it never ends

unrelated question: does steamtinkerlaunch works from within Lutris?

**Comment by frostworx** [comment]
np
I don't care about lutris. try yourself.

**Comment by folavo** [comment]
I appreciate the time you took to look in this.
keep up the good work. cheers

**Comment by sonic2kk** [comment]
Sorry to bump, I had closed issues opened for some reason and this caught my eye.

I had a read through this again and I'm not sure why it isn't working, but the Steam DRM complaint indicates to me that this may not be related to STL. There might simply be some user-end configuration that I'm not aware of.

SteamTinkerLaunch probably doesn't work through Lutris, but you might have better luck getting it to work through a Lutris community installer. If you're technically minded you could have a look at what those community scripts do and try to emulate that using SteamTInkerLaunch, but there are a lot of variables here which might be causing the problem. But this isn't the only issue related to non-steam games with STL on Steam Deck, so it might merit further investigation if more persistent issues arise. If/when I get the opportunity to test further, I'll try and see how I go.

But generally speaking, the custom command option for STL is intended for standalone executables and that's really all I've tested with.

If you encounter issues with other non-Steam games and/or you resolve this yourself, please feel free to leave a reply. If you do end up getting it resolved that'll be great news and I would love to know how. If you have issues with other non-Steam games in a similar manner, you could comment here but I would also suggest opening a new issue for those.

Basically, you're welcome to leave any updates you have on this into the future :-)
